Generalize AsymmetricUnifiedFocalLoss for multi-class and align interface

ytl0623 opened the pull request:
Fixes #8603  .

### Description

Addresses the limitations of `AsymmetricUnifiedFocalLoss` and its dependencies (`AsymmetricFocalLoss`, `AsymmetricFocalTverskyLoss`). The previous implementations were hard-coded for binary segmentation and lacked the standard `sigmoid`/`softmax`/`to_onehot_y` interface common to other MONAI losses.

### Types of changes
<!--- Put an `x` in all the boxes that apply, and remove the not applicable items -->
- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

ytl0623 committed:
Generalize AsymmetricUnifiedFocalLoss for multi-class and align interface

Signed-off-by: ytl0623 <david89062388@gmail.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Signed-off-by: ytl0623 <david89062388@gmail.com>

ytl0623 commented:
Hi @KumoLiu, @Nic-Ma  and @ericspod,

Sorry to bother.
Could you please review the PR or give some tips?

Thanks in advance!

ericspod commented on monai/losses/unified_focal_loss.py:
```diff
@@ -23,218 +23,272 @@
 
     def __init__(
         self,
-        to_onehot_y: bool = False,
```
Why is `to_onehot_y` being removed? I think users will still want this functionality so I would leave it in and add `include_background` as a new last argument. Even if we do want to remove it we need to use the [deprecation decorators to mark the argument removed](https://github.com/Project-MONAI/MONAI/blob/69f3dd26ed2a65e89ae89d951bb16f2dcb4d7c5d/monai/utils/deprecate_utils.py#L123) but still keep it for a version or two.

ericspod reviewed (commented):
Hi @ytl0623 thanks for the contribution. I had only a minor comment but please look at the greptile comments. A few look spurious they should all be resolved.

NabJa committed:
8564 fourier positional encoding (#8570)

Fixes #8564 .

### Description

Add Fourier feature positional encodings to `PatchEmbeddingBlock`. It
has been shown, that Fourier feature positional encodings are better
suited for Anistropic images and videos:
https://arxiv.org/abs/2509.02488

### Types of changes
<!--- Put an `x` in all the boxes that apply, and remove the not
applicable items -->
- [x] Non-breaking change (fix or new feature that would not break
existing functionality).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u
--net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick
--unittests --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/`
folder.

---------

Signed-off-by: NabJa <nabil.jabareen@gmail.com>
Co-authored-by: Eric Kerfoot <17726042+ericspod@users.noreply.github.com>
Signed-off-by: ytl0623 <david89062388@gmail.com>

ytl0623 committed:
Refactor parameters for UnifiedFocalLoss class

Signed-off-by: ytl0623 <david89062388@gmail.com>

ytl0623 committed:
Update unified_focal_loss.py

Signed-off-by: ytl0623 <david89062388@gmail.com>

ytl0623 committed:
test

Signed-off-by: ytl0623 <david89062388@gmail.com>

ytl0623 committed:
DCO Remediation Commit for NabJa <32510324+NabJa@users.noreply.github.com>

I, NabJa <32510324+NabJa@users.noreply.github.com>, hereby add my Signed-off-by to this commit: 76c4391ca70e9c5f1515beb00b8cf2a24dc6727b

Signed-off-by: NabJa <32510324+NabJa@users.noreply.github.com>

Signed-off-by: ytl0623 <david89062388@gmail.com>

ytl0623 committed:
Merge branch 'dev' into fix-issue-8603

ytl0623 committed:
Merge branch 'dev' into fix-issue-8603
